src/data-table/column-datetime.js

```diff
@@ -397,7 +400,7 @@ function DatetimeFilter(props) {
                     value={rangeDates[0]}
                     onChange={time =>
                       setRangeDates([
-                        applyTimeToDate(rangeDates[0], time),
+                        applyTimeToDate(rangeDates[0], utils.toJsDate(time)),
```

Comment:
This was the only place where the separation between the TimePicker and the Datepicker changes leaked out. 

The idea was to update everything in the TimePicker first, but because `applyTimeToDate` is used in this component and DatePicker, it broke down that separation. 

Once DatePicker has been fully moved over `applyTimeToDate`, I'll use utils and correctly handle all of the potential date objects that could be passed into it, but for now the easiest solution is to just convert it back into a JS Date.  And, luckily, in the default `date-fns` [date-io implementation](https://github.com/dmtrKovalenko/date-io/blob/master/packages/date-fns/src/date-fns-utils.ts#L190-L192) `toJsDate` just returns the value passed into it, and so this call won't add any extra conversions for normal users.